src/server/game/Instances/InstanceScript.cpp

```diff
@@ -55,6 +55,34 @@ void InstanceScript::SaveToDB()
     CharacterDatabase.Execute(stmt);
 }
 
+void InstanceScript::OnPlayerEnter(Player* player)
+{
+    if (!IsTwoFactionInstance())
+        return;
+
+    if (GetTeamIdInInstance() == TEAM_NEUTRAL)
+    {
+        if (Group* group = player->GetGroup())
+        {
+            if (Player* leader = ObjectAccessor::FindConnectedPlayer(group->GetLeaderGUID()))
+                _teamIdInInstance = leader->GetTeamId();
+            else
+                _teamIdInInstance = player->GetTeamId();
+        }
+        else
+            _teamIdInInstance = player->GetTeamId();
+    }
+
+    if (sWorld->getBoolConfig(CONFIG_ALLOW_TWO_SIDE_INTERACTION_GROUP))
```

Comment:
Isn't it better with an early return? That way we don't need to run the code above if 2fac is off.

Reply:
no because the code above is used for the two-faction instances such as icc, etc. regardless of double faction
it gets the team inside the instance (blizzlike)

Reply:
ah aight. Thought it was purely xfac related